src/lib/integration_test_cloud_engine/kubernetes_network.ml

```diff
@@ -388,6 +420,42 @@ module Node = struct
     send_payment ~logger t ~sender_pub_key ~receiver_pub_key ~amount ~fee
     |> Deferred.bind ~f:Malleable_error.or_hard_error
 
+  let send_snapp ~logger t ~(parties : Mina_base.Parties.t) =
+    [%log info] "Sending a snapp"
+      ~metadata:
+        [ ("namespace", `String t.namespace); ("pod_id", `String t.pod_id) ] ;
+    let open Deferred.Or_error.Let_syntax in
+    let fee_payer_pk = parties.fee_payer.data.body.public_key in
+    let fee_payer_pk_str =
+      fee_payer_pk |> Signature_lib.Public_key.Compressed.to_base58_check
+    in
+    [%log info] "send_snapp: unlocking fee payer account"
+      ~metadata:[ ("fee_payer_pk", `String fee_payer_pk_str) ] ;
+    let unlock_sender_account_graphql () =
+      let unlock_account_obj =
+        Graphql.Unlock_account.make ~password:node_password
+          ~public_key:(Graphql_lib.Encoders.public_key fee_payer_pk)
+          ()
+      in
+      exec_graphql_request ~logger ~node:t
+        ~query_name:"unlock_fee_payer_account_graphql" unlock_account_obj
+    in
+    let%bind _unlock_acct_obj = unlock_sender_account_graphql () in
+    let parties_json =
+      Mina_base.Parties.to_yojson parties |> Yojson.Safe.to_basic
+    in
+    let send_snapp_graphql () =
+      let send_snapp_obj =
+        Graphql.Send_test_snapp.make ~parties:parties_json ()
+      in
+      exec_graphql_request ~logger ~node:t ~query_name:"send_snapp_graphql"
+        send_snapp_obj
+    in
+    let%map sent_snapp_obj = send_snapp_graphql () in
+    assert (Option.is_none sent_snapp_obj#sendTestSnapp#snapp#failureReason) ;
```

Comment:
Won't this check fall when the invalid transaction test? Perhaps this could be an `Or_error.t` rather than an assertion?

Reply:
With an invalid signature, as in the test here, this line isn't reached.

I can promote the `failureReason` to a `Malleable_error.t`.